### ISSUE TITLE
The toolchain is not correctly adapted to the latest version of Eslint

### ISSUE THREAD

**Issue by juilletVent** [rspack-contrib/rsbuild-plugin-eslint#17]
## Steps to reproduce

CLI Creation Steps:

```shell
PS C:\Users\AAA\Desktop\demo> pnpm create rsbuild@latest

◆  Create Rsbuild Project
│
◇  Project name or path
│  rsbuild-project
│
◇  Select framework
│  Vanilla
│
◇  Select language
│  TypeScript
│
◇  Select additional tools (Use <space> to select, <enter> to continue)
│  Add ESLint for code linting, Add Prettier for code formatting
│
◇  Next steps ─────────╮
│                      │
│  cd rsbuild-project  │
│  pnpm install        │
│  pnpm run dev        │
│                      │
├──────────────────────╯
│
└  Done.****
```

Install Eslint Plugin dependencies and external eslint common configuration (old version of Eslint, not flattened) and import:

```shell
pnpm add @rsbuild/plugin-eslint eslint-config-gmsoft -D
```

rsbuild.config.ts：

```typescript
import { defineConfig } from "@rsbuild/core";
import { pluginEslint } from "@rsbuild/plugin-eslint";

export default defineConfig({
  plugins: [pluginEslint()],
});
```

Add the eslintrc configuration according to the Example. A version issue has been identified here; the eslint version referenced in the Example is: `^8.57.1`. The content of `.eslintrc.js` is as follows:

```javascript
module.exports = {
  extends: ["gmsoft"], // Old external common configuration, the latest eslint configuration requires flattening
};
```

Start the project:

```shell
pnpm dev

# Error message
error   Compile error:
Failed to compile, check the errors for troubleshooting.
× ESLintError: [eslint] context.getScope is not a function
  │ Occurred while linting C:\Users\97161\Desktop\demo\rsbuild-project\src\index.ts:1
  │ Rule: "@typescript-eslint/no-use-before-define"
```

## Package.json

```json
{
  "name": "rsbuild-project",
  "private": true,
  "version": "1.0.0",
  "scripts": {
    "build": "rsbuild build",
    "dev": "rsbuild dev",
    "format": "prettier --write .",
    "lint": "eslint .",
    "preview": "rsbuild preview"
  },
  "devDependencies": {
    "@eslint/js": "^9.12.0",
    "@rsbuild/core": "^1.0.19",
    "@rsbuild/plugin-eslint": "^1.0.4",
    "eslint": "^9.12.0",
    "eslint-config-gmsoft": "^2.2.5",
    "globals": "^15.11.0",
    "prettier": "^3.3.3",
    "typescript": "^5.6.3",
    "typescript-eslint": "^8.8.1"
  }
}
```

## Temporary Solution

Downgrade eslint version to `^8.57.1`:

```json
{
  "devDependencies": {
    "eslint": "^8.57.1"
  }
}
```

I tried to trace the Nodejs call stack, but @eslint/eslintrc and eslint seem to be the latest versions. Currently I cannot use the latest version of eslint, so I can only temporarily use version 8.x to solve this problem

**Comment by chenjiahan** [comment]
Does the `eslint-config-gmsoft` package support ESLint v9?

**Comment by juilletVent** [comment]
> Does the `eslint-config-gmsoft` package support ESLint v9?

You are right, the external eslint configuration I am using does not support ESLint v9. I tested it in detail and found the problem.

1. Replace with an external configuration that has been confirmed to be upgraded and supports ESLint v9 [eslint-plugin-react](https://github.com/jsx-eslint/eslint-plugin-react)
2. Start the project

The error message is as follows:

```shell
error   Compile error:
Failed to compile, check the errors for troubleshooting.
× ESLintError: [eslint] No ESLint configuration found in C:\Users\Administrator\Desktop\rspack-test\rsbuild-project\src.
```

Then, I consulted the configuration options of `eslint-rspack-plugin` and found an option regarding the configuration type: `configType`, which defaults to `"eslintrc"`. I believe this is the root of the problem. Then I returned to the README.md of `rsbuild-plugin-eslint` and found a sample configuration for ESLint v9.

So my problem has been solved, let's summarize:

1. I created a project using the official documentation.
2. The dependencies referenced in this project are all the latest versions, but the default configuration uses the eslint configuration file type `eslintrc`, which has been deprecated and cannot be used in the latest version. Additionally, the CLI tool created an Eslint configuration file of version V9 with the `flat` type, which means that the default configuration file will not be read by Eslint at all, and the CLI did not create an Eslintrc configuration, ultimately resulting in the error mentioned above.

I think we have two modification options:

Option 1: Change the default ESLint configuration file type created by the CLI to eslintrc type, because the default configuration file type of `eslint-rspack-plugin` is eslintrc.

Option 2: Change the default value of the `configType` configuration field in the `eslint-rspack-plugin` to "flat", because the default ESLint configuration created by the CLI is the latest version.

The current situation may be a bit confusing for beginners, and I hope this issue can be resolved. Wishing you a smooth workday.  🥰 🥰
